pkg/virt-launcher/virtwrap/live-migration-target.go

```diff
@@ -35,17 +37,97 @@ import (
 	"kubevirt.io/kubevirt/pkg/util/net/ip"
 	migrationproxy "kubevirt.io/kubevirt/pkg/virt-handler/migration-proxy"
 	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/cli"
 	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/converter"
 )
 
-func (l *LibvirtDomainManager) finalizeMigrationTarget(vmi *v1.VirtualMachineInstance) error {
+func (l *LibvirtDomainManager) finalizeMigrationTarget(vmi *v1.VirtualMachineInstance, options *cmdv1.VirtualMachineOptions) error {
+	interfacesToReconnect := interfacesToReconnect(options)
+	if len(interfacesToReconnect) != 0 {
+		if err := l.reconnectGuestNics(vmi, interfacesToReconnect); err != nil {
+			return err
+		}
+	}
+
 	if err := l.setGuestTime(vmi); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+func interfacesToReconnect(options *cmdv1.VirtualMachineOptions) map[string]struct{} {
+	interfaceMigrationOptions := options.InterfaceMigration
+	ifacesToRefresh := map[string]struct{}{}
+	for ifaceName, migrationOption := range interfaceMigrationOptions {
+		if migrationOption != nil && migrationOption.Method == string(v1.LinkRefresh) {
+			ifacesToRefresh[ifaceName] = struct{}{}
+		}
+	}
+	return ifacesToRefresh
+}
+
+// Set interfaces link down and up to renew DHCP leases
+func (l *LibvirtDomainManager) reconnectGuestNics(vmi *v1.VirtualMachineInstance, ifacesToRefresh map[string]struct{}) error {
+	l.domainModifyLock.Lock()
+	defer l.domainModifyLock.Unlock()
+
+	domName := api.VMINamespaceKeyFunc(vmi)
+	dom, err := l.virConn.LookupDomainByName(domName)
+	if err != nil {
+		log.Log.Object(vmi).Reason(err).Error("failed to reconnect guest interfaces")
+		return err
+	}
+	defer dom.Free()
+	xmlstr, err := dom.GetXMLDesc(0)
+	if err != nil {
+		return err
+	}
+
+	var domain api.DomainSpec
+	if err = xml.Unmarshal([]byte(xmlstr), &domain); err != nil {
+		return fmt.Errorf("parsing domain XML failed, err: %v", err)
+	}
+
+	// Look up all the interfaces and reconnect them
+	for _, iface := range domain.Devices.Interfaces {
+		if _, exist := ifacesToRefresh[iface.Alias.GetName()]; !exist {
+			continue
+		}
+
+		if err = reconnectIface(dom, iface); err != nil {
```

Comment:
Can we do this for all such interfaces once and not serially one-by-one?

Reply:
just by updating the whole domain which I find dangerous.

Reply:
OK, I thought we are already changing the whole domain. The command used is indeed per interface so this will not work.

Ref: https://libvirt.org/html/libvirt-libvirt-domain.html#virDomainUpdateDeviceFlags